packages/x-charts/src/ChartsXAxis/ChartsGroupedXAxis.tsx

```diff
@@ -0,0 +1,171 @@
+'use client';
+import * as React from 'react';
+import { ChartsXAxisProps, type AxisGroup } from '../models/axis';
+import { useDrawingArea } from '../hooks/useDrawingArea';
+import { isBandScale } from '../internals/isBandScale';
+import { useChartContext } from '../context/ChartProvider/useChartContext';
+import { TICK_LABEL_GAP, XAxisRoot } from './utilities';
+import { useTicksGrouped } from '../hooks/useTicksGrouped';
+import { useAxisProps } from './useAxisProps';
+
+const DEFAULT_GROUPING_CONFIG = {
+  tickSize: 6,
+};
+
+const getGroupingConfig = (
+  groupingConfig: { config?: Omit<AxisGroup, 'getValue'>[] },
+  groupIndex: number,
+  tickSize: number | undefined,
+) => {
+  const config = groupingConfig?.config?.[groupIndex] ?? ({} as AxisGroup);
+
+  const defaultTickSize = tickSize ?? DEFAULT_GROUPING_CONFIG.tickSize;
+  const calculatedTickSize = defaultTickSize * groupIndex * 2 + defaultTickSize;
+
+  return {
+    ...DEFAULT_GROUPING_CONFIG,
+    ...config,
+    tickSize: config.tickSize ?? calculatedTickSize,
+  };
+};
+
+/**
+ * @ignore - internal component.
+ */
+function ChartsGroupedXAxis(inProps: ChartsXAxisProps) {
+  const {
+    xScale,
+    defaultizedProps,
+    tickNumber,
+    positionSign,
+    skipAxisRendering,
+    classes,
+    Line,
+    Tick,
+    TickLabel,
+    Label,
+    axisTickLabelProps,
+    axisLabelProps,
+  } = useAxisProps(inProps);
+
+  if (!isBandScale(xScale)) {
+    throw new Error(
+      'MUI X Charts: ChartsGroupedXAxis only supports the `band` and `point` scale types.',
+    );
+  }
+
+  const {
+    position,
+    disableLine,
+    disableTicks,
+    label,
+    tickSize,
+    valueFormatter,
+    slotProps,
+    tickInterval,
+    tickPlacement,
+    tickLabelPlacement,
+    sx,
+    offset,
+    height: axisHeight,
+  } = defaultizedProps;
+
+  const groups = (defaultizedProps as { groups: AxisGroup[] }).groups;
+
+  const groupingConfig = React.useMemo(() => {
+    return {
+      getGrouping: (value: any, dataIndex: number) =>
+        groups.length > 0 ? groups.map((group) => group.getValue(value, dataIndex)) : [''],
```

Comment:
A bit weird that the returned array contains an empty string. It would feel more natural that an empty groups implies no ticks
 
```suggestion
        groups.length > 0 ? groups.map((group) => group.getValue(value, dataIndex)) : [],
```